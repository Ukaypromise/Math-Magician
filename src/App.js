/*eslint-disable*/
import React, { Component } from 'react';
import CalculatorApp from './components/Calculator';
import './App.css';

export class App extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }
render() {
return (
  <div className="App">
    <CalculatorApp />
  </div>
);
}
}
// eslint-disable for using exported App as identifier

export default App;
