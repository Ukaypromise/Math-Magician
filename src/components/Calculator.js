import React, { useState } from 'react';
import calculate from './logic/calculate';
import '../styles/calculator.css';

const Calculator = () => {
  const stateObj = {
    total: 0,
    next: null,
    operation: null,
  };

  const [state, setState] = useState(stateObj);

  const handleClick = (btn) => {
    const btnValue = btn.target.textContent;
    const newState = { ...state };
    setState(calculate(newState, btnValue));
  };

  return (
    <div className="wrapper">
      <div className="display">
        {state.total}
        {state.operation}
        {state.next}
      </div>
      <div className="keyBoard">
        <button type="button" onClick={handleClick} className="btn btn-0">
          0
        </button>
        <button type="button" onClick={handleClick} className="btn btn-1">
          1
        </button>
        <button type="button" onClick={handleClick} className="btn btn-2">
          2
        </button>
        <button type="button" onClick={handleClick} className=" btn btn-3">
          3
        </button>
        <button type="button" onClick={handleClick} className=" btn btn-4">
          4
        </button>
        <button type="button" onClick={handleClick} className="btn btn-5">
          5
        </button>
        <button type="button" onClick={handleClick} className=" btn btn-6">
          6
        </button>
        <button type="button" onClick={handleClick} className=" btn btn-7">
          7
        </button>
        <button type="button" onClick={handleClick} className="btn btn-8">
          8
        </button>
        <button type="button" onClick={handleClick} className="btn btn-9">
          9
        </button>
        <button
          type="button"
          onClick={handleClick}
          className="btn btn-Add"
        >
          +
        </button>
        <button
          type="button"
          onClick={handleClick}
          className="btn btn-Minus"
        >
          -
        </button>
        <button
          type="button"
          onClick={handleClick}
          className="btn btn-Div"
        >
          รท
        </button>
        <button
          type="button"
          onClick={handleClick}
          className="btn btn-Multiply"
        >
          x
        </button>
        <button
          type="button"
          onClick={handleClick}
          className="btn btn-Equal"
        >
          =
        </button>
        <button
          type="button"
          onClick={handleClick}
          className="btn btn-Percentage"
        >
          %
        </button>
        <button
          type="button"
          onClick={handleClick}
          className="btn btn-AllClear"
        >
          AC
        </button>
        <button
          type="button"
          onClick={handleClick}
          className="btn btn-MasMin"
        >
          +/-
        </button>
        <button
          type="button"
          onClick={handleClick}
          className="btn btn-Decimal"
        >
          .
        </button>
      </div>
    </div>
  );
};

export default Calculator;
