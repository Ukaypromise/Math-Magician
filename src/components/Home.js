import React from 'react';
import '../styles/Home.css';

const info = 'ipsum dolor sit amet consectetur, adipisicing elit. Odio error inventore perferendis dolorum rem blanditiis cupiditate impedit cum debitis nobis!';
const Home = () => (
  <div>
    <section className="Home">
      <h1>Welcome to our Page!</h1>
      <p>{info}</p>
      <p>{info}</p>
      <p>{info}</p>
    </section>
    <footer>
      <p className="footer">Designed by Promise Uka</p>
    </footer>
  </div>
);

export default Home;
