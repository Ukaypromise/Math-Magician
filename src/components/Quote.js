import React from 'react';
import '../styles/Quote.css';

const Qoute = () => {
  const quote = 'Mathematics is not about numbers equations, computations, or algorithms. It is about undersatanding.';
  const author = '-William paul Thurston';
  return (
    <section className="quote-container">
      <p>{quote}</p>
      <span>{author}</span>
    </section>
  );
};

export default Qoute;
