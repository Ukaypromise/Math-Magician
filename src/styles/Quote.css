.quote-container {
  width: 90%;
  margin: 8rem auto;
  font-size: 1.8rem;
  border-radius: 8px;
  padding: 2rem 3rem;
  margin-bottom: 1.5rem;
  border-radius: 10px;
  background: #e0e0e0;
  box-shadow: 6px -6px 8px #9d9d9d, -6px 6px 8px #ffffff;
  
}

.quote-container p {
  text-align: center;
  font-style: italic;
  font-size: 2rem;
  font-weight: bold;
  margin-bottom: 2rem;
}

.quote-container span{
    display: block;
    text-align: right;
    font-size: 1.5rem;
    font-weight: bold;
    margin-top: 1rem;
}

@media (min-width: 768px) {
 .quote-container {
  height: 300px;
  width: 60%;
 
}
.quote-container p {
   margin-top: 5rem;
   font-size: 3rem;
}

}