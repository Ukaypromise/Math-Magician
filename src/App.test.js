import { render, screen } from '@testing-library/react';
import App from './App';
import Home from "./components/Home";
import Calculator from "./components/Calculator";
import Quote from "./components/Quote";
// import userEvent from "@testing-library/user-event";

test('renders learn react link', () => {
  render(<App />);
  const linkElement = screen.getByText("Math Magicians");
  expect(linkElement).toBeInTheDocument();
});

test("Home router", () => {
  const hm = render(<Home />);
  expect(hm).toMatchSnapshot();
});
test("Calculator router", () => {
  const hm = render(<Calculator />);
  expect(hm).toMatchSnapshot();
});

test("Quote router", () => {
  const hm = render(<Quote />);
  expect(hm).toMatchSnapshot();
});
