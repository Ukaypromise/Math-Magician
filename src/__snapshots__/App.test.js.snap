// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator router 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wrapper"
      >
        <div
          class="display"
        >
          0
        </div>
        <div
          class="keyBoard"
        >
          <button
            class="btn btn-0"
            type="button"
          >
            0
          </button>
          <button
            class="btn btn-1"
            type="button"
          >
            1
          </button>
          <button
            class="btn btn-2"
            type="button"
          >
            2
          </button>
          <button
            class=" btn btn-3"
            type="button"
          >
            3
          </button>
          <button
            class=" btn btn-4"
            type="button"
          >
            4
          </button>
          <button
            class="btn btn-5"
            type="button"
          >
            5
          </button>
          <button
            class=" btn btn-6"
            type="button"
          >
            6
          </button>
          <button
            class=" btn btn-7"
            type="button"
          >
            7
          </button>
          <button
            class="btn btn-8"
            type="button"
          >
            8
          </button>
          <button
            class="btn btn-9"
            type="button"
          >
            9
          </button>
          <button
            class="btn btn-Add"
            type="button"
          >
            +
          </button>
          <button
            class="btn btn-Minus"
            type="button"
          >
            -
          </button>
          <button
            class="btn btn-Div"
            type="button"
          >
            รท
          </button>
          <button
            class="btn btn-Multiply"
            type="button"
          >
            x
          </button>
          <button
            class="btn btn-Equal"
            type="button"
          >
            =
          </button>
          <button
            class="btn btn-Percentage"
            type="button"
          >
            %
          </button>
          <button
            class="btn btn-AllClear"
            type="button"
          >
            AC
          </button>
          <button
            class="btn btn-MasMin"
            type="button"
          >
            +/-
          </button>
          <button
            class="btn btn-Decimal"
            type="button"
          >
            .
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wrapper"
    >
      <div
        class="display"
      >
        0
      </div>
      <div
        class="keyBoard"
      >
        <button
          class="btn btn-0"
          type="button"
        >
          0
        </button>
        <button
          class="btn btn-1"
          type="button"
        >
          1
        </button>
        <button
          class="btn btn-2"
          type="button"
        >
          2
        </button>
        <button
          class=" btn btn-3"
          type="button"
        >
          3
        </button>
        <button
          class=" btn btn-4"
          type="button"
        >
          4
        </button>
        <button
          class="btn btn-5"
          type="button"
        >
          5
        </button>
        <button
          class=" btn btn-6"
          type="button"
        >
          6
        </button>
        <button
          class=" btn btn-7"
          type="button"
        >
          7
        </button>
        <button
          class="btn btn-8"
          type="button"
        >
          8
        </button>
        <button
          class="btn btn-9"
          type="button"
        >
          9
        </button>
        <button
          class="btn btn-Add"
          type="button"
        >
          +
        </button>
        <button
          class="btn btn-Minus"
          type="button"
        >
          -
        </button>
        <button
          class="btn btn-Div"
          type="button"
        >
          รท
        </button>
        <button
          class="btn btn-Multiply"
          type="button"
        >
          x
        </button>
        <button
          class="btn btn-Equal"
          type="button"
        >
          =
        </button>
        <button
          class="btn btn-Percentage"
          type="button"
        >
          %
        </button>
        <button
          class="btn btn-AllClear"
          type="button"
        >
          AC
        </button>
        <button
          class="btn btn-MasMin"
          type="button"
        >
          +/-
        </button>
        <button
          class="btn btn-Decimal"
          type="button"
        >
          .
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Home router 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <section
          class="Home"
        >
          <h1>
            Welcome to our Page!
          </h1>
          <p>
            ipsum dolor sit amet consectetur, adipisicing elit. Odio error inventore perferendis dolorum rem blanditiis cupiditate impedit cum debitis nobis!
          </p>
          <p>
            ipsum dolor sit amet consectetur, adipisicing elit. Odio error inventore perferendis dolorum rem blanditiis cupiditate impedit cum debitis nobis!
          </p>
          <p>
            ipsum dolor sit amet consectetur, adipisicing elit. Odio error inventore perferendis dolorum rem blanditiis cupiditate impedit cum debitis nobis!
          </p>
        </section>
        <footer>
          <p
            class="footer"
          >
            Designed by Promise Uka
          </p>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <section
        class="Home"
      >
        <h1>
          Welcome to our Page!
        </h1>
        <p>
          ipsum dolor sit amet consectetur, adipisicing elit. Odio error inventore perferendis dolorum rem blanditiis cupiditate impedit cum debitis nobis!
        </p>
        <p>
          ipsum dolor sit amet consectetur, adipisicing elit. Odio error inventore perferendis dolorum rem blanditiis cupiditate impedit cum debitis nobis!
        </p>
        <p>
          ipsum dolor sit amet consectetur, adipisicing elit. Odio error inventore perferendis dolorum rem blanditiis cupiditate impedit cum debitis nobis!
        </p>
      </section>
      <footer>
        <p
          class="footer"
        >
          Designed by Promise Uka
        </p>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Navbar router 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="quote-container"
      >
        <p>
          Mathematics is not about numbers equations, computations, or algorithms. It is about undersatanding.
        </p>
        <span>
          -William paul Thurston
        </span>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="quote-container"
    >
      <p>
        Mathematics is not about numbers equations, computations, or algorithms. It is about undersatanding.
      </p>
      <span>
        -William paul Thurston
      </span>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Quote router 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="quote-container"
      >
        <p>
          Mathematics is not about numbers equations, computations, or algorithms. It is about undersatanding.
        </p>
        <span>
          -William paul Thurston
        </span>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="quote-container"
    >
      <p>
        Mathematics is not about numbers equations, computations, or algorithms. It is about undersatanding.
      </p>
      <span>
        -William paul Thurston
      </span>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Quote router 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="quote-container"
      >
        <p>
          Mathematics is not about numbers equations, computations, or algorithms. It is about undersatanding.
        </p>
        <span>
          -William paul Thurston
        </span>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="quote-container"
    >
      <p>
        Mathematics is not about numbers equations, computations, or algorithms. It is about undersatanding.
      </p>
      <span>
        -William paul Thurston
      </span>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
